alarm_trap_sender.cpp

```diff
@@ -50,22 +50,36 @@ AlarmFilter AlarmFilter::_instance;
 
 AlarmTrapSender AlarmTrapSender::_instance;
 
-bool ObservedAlarms::update(AlarmTableDef& alarm_table_def, const std::string& issuer)
+bool ObservedAlarms::update(const AlarmTableDef& alarm_table_def, const std::string& issuer)
 {
-  bool updated = false;
+  if (!is_active(alarm_table_def))
+  {
+    _index_to_entry[alarm_table_def.alarm_index()] = AlarmListEntry(alarm_table_def, issuer);
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
 
+bool ObservedAlarms::is_active(const AlarmTableDef& alarm_table_def)
+{
   std::map<unsigned int, AlarmListEntry>::iterator it = _index_to_entry.find(alarm_table_def.alarm_index());
-  
-  // If either the current alarm doesn't exist in the ObservedAlarms mapping
-  // or there is an entry for the current alarm in the mapping but at a
-  // different severity to the one we are currently raising the alarm with, we
-  // want to update the mapping.
+
   if ((it == _index_to_entry.end()) || (alarm_table_def.severity() != it->second.alarm_table_def().severity()))
   {
-    _index_to_entry[alarm_table_def.alarm_index()] = AlarmListEntry(alarm_table_def, issuer);
-    updated = true;
+    // Either the current alarm doesn't exist in the ObservedAlarms mapping
+    // or there is an entry for the current alarm in the mapping but at a
+    // different severity to the one we are currently raising the alarm with.
+    TRC_DEBUG("Alarm is not active at this severity");
```

Comment:
If it's active at another severity, might be worth TRC_DEBUGing that severity out.